### ISSUE TITLE
Issue with 'toast-container' class and CSS styles

### ISSUE THREAD

**Issue by alinaalam** [scttcper/ngx-toastr#492]
Hi!

I was working with toastr and was unable to apply css styles on it. There are two problems that I am facing atm: 

1. When I use @import "~ngx-toastr/toastr.css"; in my app.component.scss, it doesn't work. No styles are applied. Styles are only applied if I manually copy them from https://github.com/scttcper/ngx-toastr/blob/HEAD/src/lib/toastr.css and write '::ng-deep' with the classes

2.  The second issue is that all your styles are based on '.toast-container', when in fact, in the DOM element, a div is appended which uses the **id** 'toast-container'. If, in DOM, I change the word id to class using inspect element, I see the toast, otherwise I don't see it anywhere. In your demos, an additional 'toast-container' class gets added to your div component, but in my case, it doesn't

I am using Angular 4 and ngx-toastr@6.4.1-beta.0 in my project

![toastr](https://user-images.githubusercontent.com/23357722/42151955-184ce1ae-7df8-11e8-8def-f4658d32b487.png)

**Comment by scttcper** [comment]
style should be imported globally in either styles.css or in your angular.json

the old version you are using does use ID's https://github.com/scttcper/ngx-toastr/blob/v6/src/lib/toastr.css#L84

**Comment by alinaalam** [comment]
I am not sure why it isn't applying the styles unless I use '::ng-deep' with the styles. I guess I would just stick to this implementation then

**Comment by scttcper** [comment]
applying it to app.component styles is not what you want.

**Comment by yarrgh** [comment]
If you insist on putting it in the app.component.css, then you'd want to turn off view encapsulation for the app component so that the classes don't get encapsulated.

```ts
@NgModule({
  //...
  encapsulation: ViewEncapsulation.None
})
```

I'd avoid `::ng-deep` as it will be removed in a future version of Angular ([See Here](https://angular.io/guide/component-styles#deprecated-deep--and-ng-deep))

**Comment by alinaalam** [comment]
I want the toasts to display on all the pages, therefore, I have used the styles in app.component.scss. The functionality is written in app.component.ts. I did what you said and remove all the ::ng-deep and simply included the styles in the app.component.scss file. The styles are there, I can see them when I inspect for them, but they don't apply on the toast unless I explicitly use ::ng-deep. I think the reason is this item is created at run time, it's not their when the component is called, instead it adds in the DOM when we fire an event (when we want to display some toast), that's why the styles are not applied unless we use '::ng-deep' which is used to apply styles on items that are added in the DOM at run time
